### DISCUSSION TITLE
IBC Core  - Message Structure

### DISCUSSION THREAD

**Discussion by shreyasbhat0** [icon-project/IBC-Integration#168] [General]
Hi Everyone,

A couple of queries I have with the java implementation of Ibc Core:

MsgCreateClient Struct defined java has a ClientType field which is not present in Ibc-rs. Here instead of getting the client type from the message we have to get the same from the client state. So my question is will the client state from the other side will be having client type as a field or do we need to make use of this passed client type?

If it's present in the client state we can deserialise and get the client type.

Please let me know how we can address this part on the development side.

[Ibc-rs-ClientState](https://github.com/cosmos/ibc-rs/blob/00f9c8e95fe039fa6c234f30e6dbd4bd8f23ec71/crates/ibc/src/core/ics02_client/client_state.rs#L25) 

[ibc-rs-create-client-message](https://github.com/cosmos/ibc-rs/blob/main/crates/ibc/src/core/ics02_client/msgs/create_client.rs#L17)

**Comment by shreyasbhat0** [comment]
@AntonAndell @redlarva @MuhammedIrfan @viveksharmapoudel

**Comment by shreyasbhat0** [comment]
cc @Sid-hugobyte

**Comment by AntonAndell** [comment]
It won't have but it is only used to create client-ids, where are clientype needed?

**Reply by shreyasbhat0** [reply]
Yeah it's used to create a client id . When creating a client we expect that for generating client id .

**Reply by AntonAndell** [reply]
Hmm okey, clienstate in the ICON lightclient design wont have clientType as designed now. But since this IBC implementation will only be used with ICON for now it might be fine to only accepts a fixed client type

**Reply by viveksharmapoudel** [reply]
I guess it is better to add a separate field than keeping it in the clientState because most of the light client don't have a clientType field. These are some of the minor deviations.

**Reply by shreyasbhat0** [reply]
And I have one more doubt. When a relayer dispatches the create client message to the Archway side, will relay invoke Core Contract with the same message format you defined in Java?

I might be wrong here. Shouldn't the message should be in the same format across the IBC?

**Reply by AntonAndell** [reply]
No it does not need to be to my understanding. Creating a client on ICON will most likely be quite different than in archway
